custom_components/llama_conversation/__init__.py

```diff
@@ -279,6 +284,8 @@ async def async_process(
             
             if len(conversation) == 0:
                 conversation.append(system_prompt)
+                if remember_conversation:
+                    self.history[conversation_id] = conversation
```

Comment:
this shouldn't be necessary since you're storing the conversation on line 316

Reply:
The idea behind that is it would remember only the system prompt if both `Refresh system prompt` and `Remember conversation` were disabled instead of regenerating it again, but it would have to be `if not remember_conversation:`. which I've fixed in the PR. This should retain the small benefit that prompt caching provides if backend supports it.